RELEASES.md

```diff
@@ -1,15 +1,34 @@
 # APM Server release checklist
 
-## Before the release
+The APM Server follows the Elastic Stack release schedule and versions.
+A release starts with a Feature Freeze period, during which only bug fixes
+are allowed to be merged into the specific release branch.
+We generally follow [semver](https://semver.org/) for release versions.
+For major and minor releases a new branch is cut off the main branch.
+For patch releases, only the version on the existing major and minor version branch is updated.
 
-* Backport all relevant commits
+## Feature Freeze 
 
-  e.g. start with following command to see which files differ that are not necessarily expected:
+* Backport all relevant commits (patch releases)
+
+  We use backport labels on PRs for automated backporting of changes.
+  Everything relevant to a release should already be backported.
+  Nevertheless, a quick check to ensure nothing has been overlooked, should be done.
+  For example, use following command to see which files differ:
   `git diff elastic/main --name-only | grep -v '^vendor/' | grep -v '^docs/' | grep -v '^_beats'`
 
+* For minor releases create a new release branch and
+
+  * update versions in release branch, e.g. [#2803](https://github.com/elastic/apm-server/pull/2803/files)
+    Ensure that the `BEATS_VERSION` in the Makefile is updated. 
+  * update versions in `main` branch to next minor version, e.g. [#2804](https://github.com/elastic/apm-server/pull/2804)
+
 * Update beats
 
-  run `make update-beats` in the branch from which the new branch will be created before FF to recognize potential issues
+  When beats has merged all PRs and for minor releases created the new branch, 
+  Beats updates are automatically created for the release branch, multiple times per week. 
+  Keep an eye out for the changes they introduce. The updates are supposed to only be bug fixes.
```

Comment:
As a first-time reader, I am not sure what should be flagged while looking at changes introduced in the beats release branch, maybe we can add some examples?